.github/workflows/main.yml

```diff
@@ -0,0 +1,37 @@
+name: Main
+
+on:
+    push:
+      branches:
+        - master
+    pull_request:
+      branches:
+        - master
+
+jobs:
+    build-and-test:
+        runs-on: ${{ matrix.os }}
+        strategy:
+            fail-fast: false
+            matrix:
+                os: [ubuntu-latest]
+                python-version: [3.7, 3.8]
+        steps:
+        - uses: actions/checkout@v2
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@v2.1.2
+          with:
+            python-version: ${{ matrix.python-version }}
+        - name: Install ibis_dask with dependences
+          run: |
+            pip install -r requirements_locked.txt
+            pip install .[develop]
```

Comment:
I'd perdonally use conda and `environment.yml`. Freezing will make the CI more stable, but since devs of this backend will likely use `conda` and `environment.yml`, having it in the CI will warn you when a new package in conda is breaking something. And you can pin there packages that create too much unstability, if any.

Up to you, but that's what we do in most projects, and I think it's better to not freeze dependencies, in few months you'll be testing old stuff that it's not what your users will be installing, and you'll need to be updating this requirements_locked.txt.

Reply:
That's what we mostly do at quantopian, but I think you're right about the most common use here, I'll make that change.